spinalcordtoolbox/labels.py

```diff
@@ -267,6 +267,33 @@ def label_vertebrae(img: Image, vertebral_levels: Sequence[int] = None) -> Image
     return out
 
 
+def find_missing_label(img, ref):
+    """
+    Function that return the list of label that are present in ref and not in img.
+    This is useful to compute false negative (label that are in ref and not img) and false positive
+    (labels that are present in img and not ref)
+    :param img: source image
+    :param ref: reference image
+    :return: list of missing label
+    """
+    coordinates_input = img.getNonZeroCoordinates()
+    coordinates_ref = ref.getNonZeroCoordinates()
+
+    rounded_coord_ref_values = [np.round(c.value) for c in coordinates_ref]
+    rounded_coord_in_values = [np.round(c.value) for c in coordinates_input]
+
+    FP = [x for x in rounded_coord_in_values if x not in rounded_coord_ref_values]
+    FN = [x for x in rounded_coord_ref_values if x not in rounded_coord_in_values]
+
+    if len(FP) > 0:
+        logger.warning("False positive value for label {}".format(' '.join(map(str, FP))))
+
+    if len(FN) > 0:
+        logger.warning("False negative value for label {}".format(' '.join(map(str, FN))))
```

Comment:
I'm having some second thoughts about the terms `false positive` and `false negative` here. They have a very specific meaning (detection tasks), but this function is more general (any labeled images, not necessarily tied to detection). So the warning messages might be a bit unclear for users. (They might think "False postiive? What does that mean here, I was just trying to calculate MSE!")
 
The old messages used more general language. Maybe it would be good to preserve that. e.g. you could rename the variables `FP` -> `missing_labels_ref`, `FN` -> `missing_labels_inp`. Then you could reword the messages something like:

```suggestion
    if missing_labels_ref:
        logger.warning(f"Label mismatch: Labels {missing_labels_ref} present in input image but missing from reference image.")

    if missing_labels_inp:
        logger.warning(f"Label mismatch: Labels {missing_labels_inp} present in reference image but missing from input image.)
```